addons/sale_timesheet/controllers/pivot.py

```diff
@@ -0,0 +1,44 @@
+import json
+import math
+
+from odoo import http, _
+from odoo.http import request
+from odoo.tools import float_round
+from odoo.addons.web.controllers.pivot import TableExporter
+
+
+class PivotExporter(TableExporter):
+
+    @http.route('/web/pivot/export_xlsx', type='http', auth="user")
+    def export_xlsx(self, data, **kw):
+        jdata = self.convert_columns(data)
+        return super().export_xlsx(json.dumps(jdata), **kw)
+
+    def convert_columns(self, data):
+        """
+        Backend data are all in hours. Widget converts the uom (hours > day) only for the frontend pivot view
+        The function is to handle it error-proof for downloading xlsx feature
+        """
+
+        DAY_IN_HOURS = request.env['uom.uom'].with_context(lang='en_US').search([('name', '=', 'Hours')]).factor
```

Comment:
this is doing query for nothing in case `dict_data['model'] == 'timesheets.analysis.report'` is false. This should be done only after `if dict_data['model'] == 'timesheets.analysis.report':`

also I think you might need to do `with_context(lang=None)` because en_US could contain a translation

and third point, you should not search a uom.uom by name, because you could have 10 uom called Hours and not any of them have anything with time or with timesheet. For a record like that you always need to refer to it by XML id or by a reference saved somewhere.

as a fourth point, the data could possibly in another another unit. The unit used in the database should be the one set in `res.company().project_time_mode_id` that is set at the creation of the company:

https://github.com/odoo/odoo/blob/de02a6230d3e21e42359111c95a51485c6700773/addons/hr_timesheet/models/res_company.py#L11-L19

By default, this would be the uom.uom with xml id `uom.product_uom_hour` but this could be another uom.

Reply:
It's not a future issue, I'm saying that depending on configuration the UoM of account analytic line could be day for example (if a given configuration set days as default reference time UoM and deleted hours UoM)